services/notification_listener_test.go

```diff
@@ -276,3 +279,55 @@ func TestServices_RunLogTopic_ExpectedEventSignature(t *testing.T) {
 	expected := "0x06f4bf36b4e011a5c499cef1113c2d166800ce4013f6c2509cab1a0e92b83fb2"
 	assert.Equal(t, expected, services.RunLogTopic.Hex())
 }
+
+func TestHeadTracker_New(t *testing.T) {
+	t.Parallel()
+
+	store, cleanup := cltest.NewStore()
+	defer cleanup()
+	assert.Nil(t, store.Save(&models.BlockHeader{cltest.BigHexInt(1)}))
+	last := models.BlockHeader{cltest.BigHexInt(10)}
+	assert.Nil(t, store.Save(&last))
+	assert.Nil(t, store.Save(&models.BlockHeader{cltest.BigHexInt(2)}))
+
+	ht, err := services.NewHeadTracker(store)
+	assert.Nil(t, err)
+	assert.Equal(t, last.Number, ht.Get().Number)
+}
+
+func TestHeadTracker_Get(t *testing.T) {
+	t.Parallel()
+
+	store, cleanup := cltest.NewStore()
+	defer cleanup()
+	initial := models.BlockHeader{cltest.BigHexInt(1)}
+	assert.Nil(t, store.Save(&initial))
+
+	tests := []struct {
+		name      string
+		toSave    *models.BlockHeader
+		want      hexutil.Big
+		wantError bool
+	}{
+		// order matters
+		{"greater", &models.BlockHeader{cltest.BigHexInt(2)}, cltest.BigHexInt(2), false},
+		{"less than", &models.BlockHeader{cltest.BigHexInt(1)}, cltest.BigHexInt(2), false},
+		{"zero", &models.BlockHeader{cltest.BigHexInt(0)}, cltest.BigHexInt(2), true},
```

Comment:
Shouldn't the "less than" case err, and the "zero" case pass? The BlockHeader Number 1 is already saved in the setup, but 0 is not. Looks like they're supposed to be unique. What am I missing?

Reply:
1. Seems like the number 1 scenario silently overwrites.
2. The 0 scenario fails because in storm "id field must not be a zero value".

I'm indifferent about these edge cases. I did it to make sure we handled it graciously and had coverage. If you want something specific for 1 let me know.

Reply:
Discussed offline: I was suggesting simply removing `t.Run`.  Subtests were not a distinction I'd made, and they make debugging test failures significantly easier to track down. Sticking with it as is.

Reply:
1 seems worrisome in general, but not problematic in this case specifically.